service/src/test/java/tds/session/services/impl/SessionServiceImplTest.java

```diff
@@ -77,4 +89,145 @@ public void shouldReturnSessionAssessment() {
+        verify(mockSessionRepository).pause(mockOpenSession.getId(), "closed");
+        verify(mockSessionRepository, times(2)).findSessionById(mockOpenSession.getId());
+
+        assertThat(response.getData()).isNotNull();
+        assertThat(response.getErrors()).hasSize(0);
+        PauseSessionResponse pauseSessionResponse = response.getData().get();
+        assertThat(pauseSessionResponse.getStatus()).isEqualTo(mockUpdatedSession.getStatus());
+        assertThat(pauseSessionResponse.getSessionId()).isEqualTo(mockUpdatedSession.getId());
+        assertThat(pauseSessionResponse.getDateChanged()).isNotNull();
+        assertThat(pauseSessionResponse.getDateEnded()).isNotNull();
+    }
+
+    @Test
+    public void shouldRespondWithSessionClosedValidationErrorWhenPausingAClosedSession() {
+        long proctorId = 1L;
+        UUID sessionId = UUID.randomUUID();
+        UUID browserKey = UUID.randomUUID();
+
+        Session mockClosedSession = new Session.Builder()
+            .withId(sessionId)
+            .withStatus("closed")
+            .withDateChanged(Instant.now().minus(600000))
+            .withDateEnd(Instant.now().minus(600000))
+            .build();
+
+        PauseSessionRequest request = new PauseSessionRequest(proctorId, browserKey);
+
+        when(mockSessionRepository.findSessionById(mockClosedSession.getId())).thenReturn(Optional.of(mockClosedSession));
+        doNothing().when(mockExamService).pauseAllExamsInSession(mockClosedSession.getId());
+
+        Response<PauseSessionResponse> response = service.pause(sessionId, request);
+
+        verify(mockExamService, times(0)).pauseAllExamsInSession(mockClosedSession.getId());
+        verify(mockSessionRepository, times(0)).pause(mockClosedSession.getId(), "closed");
+        verify(mockSessionRepository, times(1)).findSessionById(mockClosedSession.getId());
+
+        assertThat(response.getData().isPresent()).isFalse();
+        assertThat(response.getErrors()).hasSize(1);
+        ValidationError error = response.getErrors()[0];
+        assertThat(error.getCode()).isEqualTo(ValidationErrorCode.PAUSE_SESSION_IS_CLOSED);
+        assertThat(error.getMessage()).isEqualTo("The session is closed");
+    }
+
+    @Test
+    public void shouldRespondWithDifferentProctorValidationErrorWhenPausingASessionWithADifferentProctorId() {
+        long proctorId = 1L;
+        UUID sessionId = UUID.randomUUID();
+        UUID browserKey = UUID.randomUUID();
+
+        Session mockClosedSession = new Session.Builder()
+            .withId(sessionId)
+            .withStatus("open")
+            .withDateEnd(Instant.now().plus(600000))
+            .withProctorId(5L)
+            .build();
+
+        PauseSessionRequest request = new PauseSessionRequest(proctorId, browserKey);
+
+        when(mockSessionRepository.findSessionById(mockClosedSession.getId())).thenReturn(Optional.of(mockClosedSession));
```

Comment:
Suggest making the changes I mentioned above to any test where the session is not valid.